src/SDL_net.c

```diff
@@ -0,0 +1,1662 @@
+#include "SDL3_net/SDL_net.h"
+
+#ifdef __WINDOWS__
+#define WIN32_LEAN_AND_MEAN 1
+#define _WIN32_WINNT 0x0600  /* we need APIs that didn't arrive until Windows Vista. */
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#include <iphlpapi.h>
+typedef SOCKET Socket;
+typedef int SockLen;
+typedef SOCKADDR_STORAGE AddressStorage;
+
+static int write(SOCKET s, const char *buf, size_t count) {
+    return send(s, buf, count, 0);
```

Comment:
```suggestion
static int write(SOCKET s, const void *buf, size_t count) {
    return send(s, (const char *)buf, count, 0);
```

because of these warnings on msys2:
```
 D:/a/SDL_net/SDL_net/src/SDL_net.c:1051:50: warning: passing 'Uint8 *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        const int bw = (int) write(sock->handle, sock->pending_output_buffer, sock->pending_output_len);
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~
D:/a/SDL_net/SDL_net/src/SDL_net.c:13:40: note: passing argument to parameter 'buf' here
static int write(SOCKET s, const char *buf, size_t count) {
                                       ^
D:/a/SDL_net/SDL_net/src/SDL_net.c:1417:43: warning: passing 'Uint8[65536]' (aka 'unsigned char[65536]') to parameter of type 'char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
    const int br = recvfrom(sock->handle, sock->recv_buffer, sizeof (sock->recv_buffer), 0, (struct sockaddr *) &from, &fromlen);
                                          ^~~~~~~~~~~~~~~~~
D:/a/_temp/msys64/clang64/include/winsock2.h:1029:58: note: passing argument to parameter 'buf' here
  WINSOCK_API_LINKAGE int WSAAPI recvfrom(SOCKET s,char *buf,int len,int flags,struct sockaddr *from,int *fromlen);
                                                         ^
```

https://github.com/libsdl-org/SDL_net/actions/runs/6328625286/job/17187212344#step:8:24